### ISSUE TITLE
setup.sh - Failed to install provider, darwin_arm64

### ISSUE THREAD

**Issue by wowpetr** [hashicorp/tfc-getting-started#30]
Hi, I'm using Terraform v1.3.7. When I try to run ./scripts/setup.sh I get the following error (executing `terraform providers lock -platform=linux_amd64` fixes it though):
```
--------------------------------------------------------------------------
Getting Started with Terraform Cloud
-------------------------------------------------------------------------

Terraform Cloud offers secure, easy-to-use remote state management and allows
you to run Terraform remotely in a controlled environment. Terraform Cloud runs
can be performed on demand or triggered automatically by various events.

This script will set up everything you need to get started. You'll be
applying some example infrastructure - for free - in less than a minute.

First, we'll do some setup and configure Terraform to use Terraform Cloud.

Press any key to continue (ctrl-c to quit):

Creating an organization and workspace...

Writing remote backend configuration to backend.tf...

========================================================================

Ready to go; the example configuration is set up to use Terraform Cloud!

An example workspace named 'getting-started' was created for you.
You can view this workspace in the Terraform Cloud UI here:
https://app.terraform.io/app/example-org-40f7b3/workspaces/getting-started

Next, we'll run 'terraform init' to initialize the backend and providers:

$ terraform init

Press any key to continue (ctrl-c to quit):


Initializing the backend...

Successfully configured the backend "remote"! Terraform will automatically
use this backend unless the backend configuration changes.

Initializing provider plugins...
- Finding latest version of hashicorp/fakewebservices...
- Installing hashicorp/fakewebservices v0.2.3...
- Installed hashicorp/fakewebservices v0.2.3 (unauthenticated)

Terraform has created a lock file .terraform.lock.hcl to record the provider
selections it made above. Include this file in your version control repository
so that Terraform can guarantee to make the same selections by default when
you run "terraform init" in the future.

╷
│ Warning: Incomplete lock file information for providers
│
│ Due to your customized provider installation methods, Terraform was forced to calculate lock file checksums locally for
│ the following providers:
│   - hashicorp/fakewebservices
│
│ The current .terraform.lock.hcl file only includes checksums for darwin_arm64, so Terraform running on another platform
│ will fail to install these providers.
│
│ To calculate additional checksums for another platform, run:
│   terraform providers lock -platform=linux_amd64
│ (where linux_amd64 is the platform to generate)
╵

Terraform has been successfully initialized!

You may now begin working with Terraform. Try running "terraform plan" to see
any changes that are required for your infrastructure. All Terraform commands
should now work.

If you ever set or change modules or backend configuration for Terraform,
rerun this command to reinitialize your working directory. If you forget, other
commands will detect it and remind you to do so if necessary.

...

========================================================================

Now it’s time for 'terraform plan', to see what changes Terraform will perform:

$ terraform plan

Press any key to continue (ctrl-c to quit):

Running plan in the remote backend. Output will stream here. Pressing Ctrl-C
will stop streaming the logs, but will not stop the plan running remotely.

Preparing the remote plan...

To view this run in a browser, visit:
https://app.terraform.io/app/example-org-40f7b3/getting-started/runs/run-aVeNZTacX83kT2Zt

Waiting for the plan to start...

Terraform v1.3.7
on linux_amd64
Initializing plugins and modules...

Initializing Terraform Cloud...

Initializing provider plugins...
- Reusing previous version of hashicorp/fakewebservices from the dependency lock file
- Installing hashicorp/fakewebservices v0.2.3...
╷
│ Error: Failed to install provider
│
│ Error while installing hashicorp/fakewebservices v0.2.3: the current
│ package for registry.terraform.io/hashicorp/fakewebservices 0.2.3 doesn't
│ match any of the checksums previously recorded in the dependency lock file;
│ for more information:
│ https://www.terraform.io/language/provider-checksum-verification
╵

Operation failed: failed running terraform init (exit 1)
```

**Comment by laurenolivia** [comment]
@wowpetr Thank you for submitting an issue. Our team will look into this.

**Comment by wowpetr** [comment]
This issue isn't the script issue, but the Terraform itself https://github.com/hashicorp/terraform/issues/28041 and it's going to be fixed in Terraform v1.4 release. So I'm closing this issue.
